### ISSUE TITLE
[Feature Request] TSC diagnostics live feedback

### ISSUE THREAD

**Issue by barrett-ruth** [jose-elias-alvarez/null-ls.nvim#940]
### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues) and there are no existing ones with the same request.

### Feature description

You probably implemented TSC with diagnostics on save for a reason, but just asking it it's possible to configure the source with live feedback.

### Help

Yes, but I don't know how to start. I would need guidance

**Comment by jose-elias-alvarez** [comment]
Maybe I'm misunderstanding the request, but wouldn't you be able to get this from setting up [tsserver](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#tsserver)? The only reason we have a `tsc` source at all is because the server doesn't support getting all diagnostics for a given project.

**Comment by barrett-ruth** [comment]
I guess I just thought using tsc would be a lot faster as I don't use much of tsserver's capabilities and it seems to have a lot of other things. But I'll just use it if that's the case.

**Comment by jose-elias-alvarez** [comment]
It's theoretically possible to do this via null-ls, but the language server will be quite a bit faster. null-ls would have to run a fresh instance of `tsc` on every change, whereas the server will start once and update each file's content as you edit.

> Also, can you elaborate on what "server doesn't support getting all diagnostics for a given project" means? Thanks.

`tsserver` will show you diagnostics for files that you've opened in Neovim, but it doesn't know anything about issues in files that you haven't opened. That's why the null-ls source is useful - if you make a change in one file that creates issues in other files, you can use the source to, say, load all of the files with errors into the quickfix list with `:lua vim.diagnostic.setqflist()` (I use this workflow a lot).

**Comment by barrett-ruth** [comment]
Understood. Thanks!

**Comment by barrett-ruth** [comment]
Sorry for the ping, but I just noticed the significance of your comment - that is, having some way for an LSP to detect errors in files that are NOT opened. Have you personally found a way to replicate this behavior with `tsc` and `tsserver` with lua?

**Comment by barrett-ruth** [comment]
Ok, last question - I hope you don't mind because this is a bit off topic from the original issue, but still conceptually related.

I checked out your dotfiles and it seems like you're using selene with lua-language-server for a similar reason.
I installed selene, set it up in a directory so both it and lua-language-server are running.

In that directory I have two files with errors. Yet, selene, which sounds like it should pick up errors on both files, doesn't. I have to open the other file in a buffer *and then* I get errors for both buffers. In this case, it seems like selene is completely useless - it's not like its catching any extra errors, or anything.

Am I setting selene up wrong? Or is this expected behavior? I want to be able to utilize null-ls in scenarios like this.
